core/src/main/scala/org/apache/spark/rapids/tool/benchmarks/BenchmarkBase.scala

```diff
@@ -47,6 +51,69 @@ abstract class BenchmarkBase {
   def prefix: String = "rapids-tools-benchmark"
   def suffix: String = ""
 
+  /**
+   * Add a benchmark case to the suite
+   * @param name Name of the benchmark case
+   * @param numIters Number of iterations to run
+   * @param f Function to run
+   */
+  def addCase(name: String, numIters: Int = 0)(f: Int => Unit): Unit = {
+    benchmark.get.addCase(name, numIters)(f)
+  }
+
+  /**
+   * Method to trigger the benchmarker cases run method
+   */
+  def run(): Unit = {
+    // Getting correct output stream
+    val printStream = output.getOrElse(System.out)
+    // Running the underlying benchmark
+    val results: mutable.ArrayBuffer[Benchmark.Result] = benchmark.get.run()
```

Comment:
If we modify the return type of `run()` to `Seq` then we should modify here as well.